src/connectors/menu/connectMenu.js

```diff
@@ -232,22 +232,9 @@ export default function connectMenu(renderFn, unmountFn = noop) {
       dispose({ state }) {
         unmountFn();
 
-        let nextState = state;
-
-        if (state.isHierarchicalFacetRefined(attribute)) {
```

Comment:
The condition prevent the function `removeHierarchicalFacetRefinement` to throw an error if the no refinement are present (https://github.com/algolia/instantsearch.js/pull/2610). It worth adding a test where we remove the widget without any refinement to ensure that we don't break this behaviour.